### PR TITLE
coll: Message combining algorithms for neighborhood collectives

### PR DESCRIPTION
**Pull request by raffenet**
## Pull Request Description

Rebased version of #2631.

do-not-merge:

[ ] Concensus
[ ] Copyright and code clean-ups

## Author Checklist
* [ ] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [ ] Passes tests (included warning check)
* [ ] Confirm whitespace/style checkers are happy (or has a good reason for being bad)
* [ ] Commits are self-contained and do not do two things at once
* [ ] Remove xfail from the test suite when fixing a test
* [ ] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [ ] Add comments such that someone without knowledge of the code could understand
* [ ] Add Devel Docs in the `doc/` directory for any new code design

### CODE REVIEW

**Comment by hzhou** [comment]
test:jenkins/warnings/quick

**Comment by hzhou** [comment]
test:jenkins/warnings/quick

**Comment by hzhou** [comment]
test:jenkins/warnings/quick

**Comment by hzhou** [comment]
test:jenkins/warnings/quick

**Comment by hzhou** [comment]
test:jenkins/warnings/quick

**Comment by hzhou** [comment]
test:jenkins/warnings/quick

**Comment by hzhou** [comment]
test:jenkins/warnings/quick

**Comment by hzhou** [comment]
test:jenkins/warnings/quick

**Comment by hzhou** [comment]
test:jenkins/warnings/quick

**Comment by hzhou** [comment]
test:jenkins/warnings/quick

**Comment by hzhou** [comment]
test:jenkins/warnings/quick

**Comment by hzhou** [comment]
test:jenkins/warnings/quick

**Comment by hzhou** [comment]
test:jenkins/warnings/quick

**Comment by hzhou** [comment]
test:jenkins/warnings/quick

**Comment by hzhou** [comment]
test:jenkins/warnings/quick

**Comment by hzhou** [comment]
test:jenkins/warnings/quick

**Comment by hzhou** [comment]
test:jenkins/ch3/tcp

File: src/include/mpir_topo.h [hunk 1]
```diff
@@ -5,6 +5,95 @@
  *
  */
 
+/*---------------------------------------------------------------------*/
+/*     (C) Copyright 2017 Parallel Processing Research Laboratory      */
+/*                   Queen's University at Kingston                    */
+/*                Neighborhood Collective Communication                */
+/*                    Seyed Hessamedin Mirsadeghi                      */
+/*---------------------------------------------------------------------*/
```
Comment (wesbland) [inline src/include/mpir_topo.h]: Given the way we do copyrights now (per #4399). Maybe this needs to change? (and all the other places in this PR of course)

Comment (hzhou) [reply src/include/mpir_topo.h]: Yes. :) However, there is no agenda to merge this anytime soon and there are many changes will be needed if we start to consider a merge.

Comment (wesbland) [reply src/include/mpir_topo.h]: No problem. I'm just working my way through the PR backlog in light of the recent discussions on Slack. 😄

**Comment by pavanbalaji** [comment]
@raffenet   What's the status of this PR?

**Comment by raffenet** [comment]
These algorithms are not useful without the persistent collectives API. It can be closed.

**Pull request closed without merging**